eth/db/journal.py

```diff
@@ -384,6 +421,8 @@ def diff(self) -> DBDiff:
                     continue
```

Comment:
There's a comment above this line which I think is no longer correct: 

`# This is purely for performance, not correctness.`

That changesets are iterated in reverse means that `ERASE_CREATED_ENTRY` has a chance to set `visited_keys` and prevent the earlier operation which set this key from showing up in the diff.